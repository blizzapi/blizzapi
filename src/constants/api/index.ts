export { hosts as regionHosts } from "./regionHosts";
