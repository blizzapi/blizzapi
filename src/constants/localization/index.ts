export { defaultLocales } from './defaultLocales';
export { locales } from './locales';
export { regions } from './regions';
export { sc2Realms } from './sc2Realms';
