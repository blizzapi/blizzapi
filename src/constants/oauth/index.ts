import { oAuthHosts } from "./OAuthHosts";
import { RegionIdProperties, RegionId, Uri } from "../../types";

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> =>
  Object.freeze({
    [RegionId.us]: `${oAuthHosts[RegionId.us]}${endpoint}`,
    [RegionId.eu]: `${oAuthHosts[RegionId.eu]}${endpoint}`,
    [RegionId.kr]: `${oAuthHosts[RegionId.kr]}${endpoint}`,
    [RegionId.cn]: `${oAuthHosts[RegionId.cn]}${endpoint}`,
  });

const authorizeEndpoint = "/oauth/authorize";
const tokenEndpoint = "/oauth/token";
const checkTokenEndpoint = "/oauth/check_token?token=";

export const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
export const tokenUris = getRegionPropertyArray(tokenEndpoint);
export const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);
