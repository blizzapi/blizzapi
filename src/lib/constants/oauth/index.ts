import { RegionIdProperties } from '../../../../@interfaces';
import { Uri } from '../../../../@types';
import OAuthHosts from './OAuthHosts';

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> => ({
  1: `${OAuthHosts[1]}${endpoint}`,
  2: `${OAuthHosts[2]}${endpoint}`,
  3: `${OAuthHosts[3]}${endpoint}`,
  5: `${OAuthHosts[5]}${endpoint}`,
});

const authorizeEndpoint = '/oauth/authorize';
const tokenEndpoint = '/oauth/token';
const checkTokenEndpoint = '/oauth/check_token?token=';

const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
const tokenUris = getRegionPropertyArray(tokenEndpoint);
const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);

export default {
  authorizeUris: Object.freeze(authorizeUris),
  tokenUris: Object.freeze(tokenUris),
  checkTokenUris: Object.freeze(checkTokenUris),
}
