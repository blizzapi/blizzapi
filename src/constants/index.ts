/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import {
  ConstantKey,
  DefaultLocaleIndex,
  LocaleArray,
  RegionIdProperties,
  RegionNameArray,
  Sc2RealmArray,
} from "../types";
import * as localization from "./localization";
import * as api from "./api";
import * as oauth from "./oauth";

export const constants = {
  [ConstantKey.REGIONS]:
    localization.regions as RegionIdProperties<RegionNameArray>,
  [ConstantKey.LOCALES]:
    localization.locales as RegionIdProperties<LocaleArray>,
  [ConstantKey.DEFAULT_LOCALES]:
    localization.defaultLocales as RegionIdProperties<DefaultLocaleIndex>,
  [ConstantKey.SC2_REALMS]:
    localization.sc2Realms as RegionIdProperties<Sc2RealmArray>,
  [ConstantKey.REGION_API_HOSTS]: api.regionHosts,
  [ConstantKey.OAUTH_AUTHORIZE_URIS]: oauth.authorizeUris,
  [ConstantKey.OAUTH_TOKEN_URIS]: oauth.tokenUris,
  [ConstantKey.OAUTH_CHECK_TOKEN_URIS]: oauth.checkTokenUris,
};
