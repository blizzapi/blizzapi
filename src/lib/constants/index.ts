/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import * as localization from './localization';
import * as api from './api';
import * as oauth from './oauth';

const CONSTANTS = {
  REGIONS: localization.regions,
  LOCALES: localization.locales,
  DEFAULT_LOCALES: localization.defaultLocales,
  SC2_REALMS: localization.sc2Realms,
  REGION_API_HOSTS: api.regionHosts,
  OAUTH_AUTHORIZE_URIS: oauth.authorizeUris,
  OAUTH_TOKEN_URIS: oauth.tokenUris,
  OAUTH_CHECK_TOKEN_URIS: oauth.checkTokenUris,
};

export default CONSTANTS;
