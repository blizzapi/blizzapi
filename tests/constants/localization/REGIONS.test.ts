import constants from '../../../src/lib/constants';

const { REGIONS } = constants;

describe('REGIONS', () => {
  test('should be defined', () => {
    expect(REGIONS).toBeDefined();
  });

  test('should be frozen', () => {
    expect(Object.isFrozen(REGIONS)).toBe(true);
  });
  test('should match snapshot', () => {
    expect(REGIONS).toMatchSnapshot();
  });
});