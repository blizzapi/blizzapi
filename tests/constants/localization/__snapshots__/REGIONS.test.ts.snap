// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REGIONS should match snapshot 1`] = `
Object {
  "1": Array [
    "us",
  ],
  "2": Array [
    "eu",
  ],
  "3": Array [
    "kr",
    "tw",
  ],
  "5": Array [
    "cn",
  ],
}
`;
